### PR TITLE
pull mode register add check cluster exists

### PR DESCRIPTION
**Pull request by yanfeng1992**
Signed-off-by: huangyanfeng <huangyanfeng1992@gmail.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

pull mode register add check cluster exists

**Which issue(s) this PR fixes**:
Fixes #

https://github.com/karmada-io/karmada/issues/4527

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by yanfeng1992** [comment]
Please take a look, thanks @chaosi-zju @lonelyCZ

**Comment by chaosi-zju** [comment]
So your solution for #4527 is to give a clear error report if the cluster exists?

**Comment by yanfeng1992** [comment]
> So your solution for #4527 is just give a clear error report if the cluster exists?

The reason why #4527 appears is because the pull cluster allows cluster registers with duplicate names. I think the current check can avoid the occurrence of #4527. @chaosi-zju 

cc @RainbowMango Do you have any suggestions?

File: pkg/karmadactl/register/register.go [hunk 1]
```diff
@@ -364,6 +365,18 @@ func (o *CommandRegisterOption) Run(parentCommand string) error {
...
+	fmt.Println("[karmada-agent-start] Waiting to check cluster exists")
+	karmadaClient, err := ToKarmadaClient(karmadaAgentCfg)
+	if err != nil {
+		return err
+	}
+	_, exist, err := karmadautil.GetClusterWithKarmadaClient(karmadaClient, o.ClusterName)
+	if err != nil {
+		return err
+	} else if exist {
+		return fmt.Errorf("cluster %s already exists", o.ClusterName)
```
Comment (RainbowMango) [inline pkg/karmadactl/register/register.go:377]:
```suggestion
		return fmt.Errorf("failed to register as cluster with name %s already exists", o.ClusterName)
```
How about this?

Comment (yanfeng1992) [reply pkg/karmadactl/register/register.go]: I think it's ok

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

/remove-kind bug
/kind cleanup

**Pull request merged**